src/compiler/util.h

```diff
@@ -122,6 +122,30 @@ class PathBuilder {
     join(segment4);
   }
 
+  // Joins the given sub_path to this builder.
+  // The sub_path must use slashes as path separators.
+  void joinSlashPath(const std::string& sub_path) {
+    if (sub_path.empty()) return;
+    size_t last_start = 0;
+    size_t pos = 0;
+    if (sub_path[0] == '/') {
+      join("/");
+      last_start = 1;
```

Comment:
It feels a bit odd that `pos` is zero after this. You might as well **not** set `last_start` to 1 here, even though that would be (perhaps) even odder.

Reply:
Completely rewritten.